src/canister_tests/src/tests.rs

```diff
@@ -955,6 +956,262 @@ mod http_tests {
         }
         Ok(())
     }
+
+    /// Verifies that all expected metrics are available via the HTTP endpoint.
+    #[test]
+    fn ii_canister_serves_http_metrics() -> Result<(), CallError> {
+        let metrics = vec![
+            "internet_identity_user_count",
+            "internet_identity_min_user_number",
+            "internet_identity_max_user_number",
+            "internet_identity_signature_count",
+            "internet_identity_stable_memory_pages",
+            "internet_identity_last_upgrade_timestamp",
+            "internet_identity_inflight_challenges",
+            "internet_identity_users_in_registration_mode",
+        ];
+        let env = StateMachine::new();
+        env.advance_time(Duration::from_secs(300)); // advance time to see it reflected on the metrics endpoint
+        let canister_id = framework::install_ii_canister(&env, framework::II_WASM.clone());
+
+        let metrics_body = flows::get_metrics(&env, canister_id);
+        for metric in metrics {
+            let (_, metric_timestamp) = framework::parse_metric(&metrics_body, metric);
+            assert_eq!(
+                metric_timestamp,
+                env.time(),
+                "metric timestamp did not match state machine time"
+            )
+        }
+        Ok(())
+    }
+
+    /// Verifies that the metrics list the expected user range.
+    #[test]
+    fn metrics_should_list_expected_user_range() -> Result<(), CallError> {
+        let env = StateMachine::new();
+        let canister_id = framework::install_ii_canister(&env, framework::II_WASM.clone());
+
+        let metrics = flows::get_metrics(&env, canister_id);
+
+        let (min_user_number, _) =
+            framework::parse_metric(&metrics, "internet_identity_min_user_number");
+        let (max_user_number, _) =
+            framework::parse_metric(&metrics, "internet_identity_max_user_number");
+        assert_eq!(min_user_number, 10_000);
+        assert_eq!(max_user_number, 3_784_872);
+        Ok(())
+    }
+
+    /// Verifies that the user count metric is updated correctly.
+    #[test]
+    fn metrics_user_count_should_increase_after_register() -> Result<(), CallError> {
+        let env = StateMachine::new();
+        let canister_id = framework::install_ii_canister(&env, framework::II_WASM.clone());
+
+        for count in 0..2 {
+            let metrics = flows::get_metrics(&env, canister_id);
+
+            let (user_count, _) = framework::parse_metric(&metrics, "internet_identity_user_count");
+            assert_eq!(user_count, count);
+
+            flows::register_anchor(&env, canister_id);
+        }
+        Ok(())
+    }
+
+    /// Verifies that the signature count metric is updated correctly.
+    #[test]
+    fn metrics_signature_count_should_increase_after_prepare_delegation() -> Result<(), CallError> {
```

Comment:
```suggestion
    fn metrics_signature_count() -> Result<(), CallError> {
```

since you're also checking for pruning, you're not only checking that it gets increases, but also decreased